plugin/src/main/kotlin/tanvd/grazi/language/LangDownloader.kt

```diff
@@ -0,0 +1,174 @@
+package tanvd.grazi.language
+
+import com.intellij.codeInsight.daemon.DaemonCodeAnalyzer
+import com.intellij.openapi.progress.impl.CoreProgressManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.ui.Messages
+import com.intellij.util.download.DownloadableFileService
+import com.intellij.util.lang.UrlClassLoader
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils
+import org.eclipse.aether.RepositorySystem
+import org.eclipse.aether.RepositorySystemSession
+import org.eclipse.aether.artifact.Artifact
+import org.eclipse.aether.artifact.DefaultArtifact
+import org.eclipse.aether.collection.CollectRequest
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory
+import org.eclipse.aether.graph.Dependency
+import org.eclipse.aether.graph.DependencyNode
+import org.eclipse.aether.graph.Exclusion
+import org.eclipse.aether.impl.DefaultServiceLocator
+import org.eclipse.aether.repository.LocalRepository
+import org.eclipse.aether.repository.RemoteRepository
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory
+import org.eclipse.aether.spi.connector.transport.TransporterFactory
+import org.eclipse.aether.transport.file.FileTransporterFactory
+import org.eclipse.aether.transport.http.HttpTransporterFactory
+import org.eclipse.aether.util.artifact.JavaScopes
+import org.languagetool.Language
+import org.languagetool.Languages
+import org.slf4j.LoggerFactory
+import tanvd.grazi.GraziConfig
+import tanvd.grazi.GraziLibResolver
+import tanvd.grazi.GraziPlugin
+import tanvd.grazi.JreProxySelector
+import tanvd.grazi.ide.ui.components.dsl.msg
+import java.io.File
+import java.net.URL
+import java.nio.file.Paths
+
+object LangDownloader {
+    private val logger = LoggerFactory.getLogger(LangDownloader::class.java)
+
+    private val repository: RepositorySystem by lazy {
+        with(MavenRepositorySystemUtils.newServiceLocator()) {
+            addService(RepositoryConnectorFactory::class.java, BasicRepositoryConnectorFactory::class.java)
+            addService(TransporterFactory::class.java, FileTransporterFactory::class.java)
+            addService(TransporterFactory::class.java, HttpTransporterFactory::class.java)
+            setErrorHandler(object : DefaultServiceLocator.ErrorHandler() {
+                override fun serviceCreationFailed(type: Class<*>?, impl: Class<*>?, exception: Throwable?) {
+                    if (exception != null) {
```

Comment:
`exception?.let {  throw RuntimeException(it) }` - oneliner)

Reply:
Done